tools/cert-tool/cmd/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/plgd-dev/kit/v2/security/generateCertificate"
+
+	flags "github.com/jessevdk/go-flags"
+	"github.com/plgd-dev/kit/v2/security"
+)
+
+type Options struct {
+	Command struct {
+		GenerateRootCA         bool   `long:"generateRootCA"`
+		GenerateIntermediateCA bool   `long:"generateIntermediateCA"`
+		GenerateCert           bool   `long:"generateCertificate"`
+		GenerateIdentity       string `long:"generateIdentityCertificate" description:"deviceID"`
+		GenerateIdentityCsr    string `long:"generateIdentityCsr" description:"deviceID"`
+	} `group:"Command" namespace:"cmd"`
+	Certificate generateCertificate.Configuration `group:"Certificate" namespace:"cert"`
+	OutCert     string                            `long:"outCert" default:"cert.pem"`
+	OutKey      string                            `long:"outKey" default:"cert.key"`
+	OutCsr      string                            `long:"outCsr" default:"req.csr"`
+	SignerCert  string                            `long:"signerCert"`
+	SignerKey   string                            `long:"signerKey"`
+}
+
+func pemBlockForKey(k *ecdsa.PrivateKey) (*pem.Block, error) {
+	b, err := x509.MarshalECPrivateKey(k)
+	if err != nil {
+		return nil, err
+	}
+	return &pem.Block{Type: "EC PRIVATE KEY", Bytes: b}, nil
+}
+
+func main() {
+	var opts Options
+	parser := flags.NewParser(&opts, flags.Default)
+	_, err := parser.Parse()
+
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(2)
+	}
+
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var cert []byte
+	switch {
```

Comment:
SonarCloud might complain that this is too complex of a function

Reply:
fixed